monkey/infection_monkey/credential_collectors/mimikatz_collector/mimikatz_credential_collector.py

```diff
@@ -11,12 +13,32 @@
 logger = logging.getLogger(__name__)
 
 
+MIMIKATZ_CREDENTIAL_COLLECTOR_TAG = "MimikatzCredentialsStolen"
+T1003_ATTACK_TECHNIQUE_TAG = "T1003"
+T1005_ATTACK_TECHNIQUE_TAG = "T1005"
+
+MIMIKATZ_EVENT_TAGS = {
+    MIMIKATZ_CREDENTIAL_COLLECTOR_TAG,
+    T1003_ATTACK_TECHNIQUE_TAG,
+    T1005_ATTACK_TECHNIQUE_TAG,
+}
+
+
 class MimikatzCredentialCollector(ICredentialCollector):
+    def __init__(self, event_queue: IEventQueue):
+        self._event_queue = event_queue
+
     def collect_credentials(self, options=None) -> Sequence[Credentials]:
         logger.info("Attempting to collect windows credentials with pypykatz.")
         windows_credentials = pypykatz_handler.get_windows_creds()
+
         logger.info(f"Pypykatz gathered {len(windows_credentials)} credentials.")
-        return MimikatzCredentialCollector._to_credentials(windows_credentials)
+
+        collected_credentials = MimikatzCredentialCollector._to_credentials(windows_credentials)
+
+        self._publish_credentials_stolen_event(collected_credentials)
```

Comment:
Similar to `SSHCredentialCollector` each credential collected represents a single event. In the case of pypykatz we can have multiple logon session that each of them have different credentials that are collected. We need to publish for each one of those sessions.

Reply:
Pypykatz collects all the credentials in one go (see [this](https://github.com/guardicore/monkey/blob/develop/monkey/infection_monkey/credential_collectors/mimikatz_collector/pypykatz_handler.py#L32)) which makes it seem more like a single event to me.